packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/CSharpType.cs

```diff
@@ -125,13 +125,18 @@ private static void ValidateArguments(Type type, IReadOnlyList<CSharpType> argum
             }
         }
 
-        internal CSharpType(TypeProvider implementation, bool isValueType = false, bool isEnum = false, bool isNullable = false, IReadOnlyList<CSharpType>? arguments = null, CSharpType? declaringType = null, string? ns = null, string? name = null)
+        internal CSharpType(TypeProvider implementation, IReadOnlyList<CSharpType>? arguments = null, bool isNullable = false)
```

Comment:
it does not make much sense that a type creates from a `TypeProvider` still needs these extra information to build, especially we could get these extra information from the `TypeProvider` instance.
Needing these extra parameters creates troubles when we try to instantiate a new type from a provider.